### PR TITLE
Install RKE2 by default

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by furkatgofurov7** [comment]
We have to make sure the uninstall process is not broken with this change, which IMO would probably be the case, since we are currently deleting deployment/webhooks specific to Kubeadm

File: charts/rancher-turtles/templates/rke2-bootstrap.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{{- if and (index .Values "cluster-api-operator" "cluster-api" "enabled") (index .Values "cluster-api-operator" "cluster-api" "rke2" "enabled") }}
```
Comment (furkatgofurov7) [inline charts/rancher-turtles/templates/rke2-bootstrap.yaml:1]: This was mainly based of off the core-yaml file and that has changed in https://github.com/rancher/turtles/pull/458/files, so this and `rke2-controlplane.yaml` files also needs the same changes wrt namespace check and hook

File: charts/rancher-turtles/questions.yml [hunk 2]
```diff
@@ -45,3 +45,8 @@ questions:
     description: "Flag to enable or disable installation of cert-manager. If set to false then you will need to install cert-manager manually"
     label: "Enable Cert Manager"
     type: boolean
+  - variable: cluster-api-operator.cluster-api.rke2.enabled
+    default: "true"
+    description: "Flag to enable or disable installation of the RKE2 provider for Cluster API. By default this is enabled."
+    label: "Enable Cert Manager"
```
Comment (furkatgofurov7) [inline charts/rancher-turtles/questions.yml:51]:
```suggestion
    label: "Enable RKE2 Provider"
```

File: charts/rancher-turtles/templates/post-delete-job.yaml [hunk 3]
```diff
@@ -122,7 +126,7 @@ spec:
         - delete
         - deployments.apps/capi-controller-manager
         - -n
-        - capi-system
+        - {{ index .Values "cluster-api-operator" "cluster-api" "core" "namespace" }}
```
Comment (furkatgofurov7) [inline charts/rancher-turtles/templates/post-delete-job.yaml:129]: can we do the same for kubeadm-cpl/bootstrap namespaces (lines 138,147)?

File: charts/rancher-turtles/questions.yml (hunk 2 above)
Comment (Danil-Grigorev) [inline charts/rancher-turtles/questions.yml:51]:
```suggestion
    label: "Enable RKE2 Provider"
```

File: charts/rancher-turtles/templates/post-delete-job.yaml (hunk 3 above)
Comment (alexander-demicev) [reply charts/rancher-turtles/templates/post-delete-job.yaml:129]: I don't know hot to handle it, we removed values for kubeadm namespace

Comment (furkatgofurov7) [reply charts/rancher-turtles/templates/post-delete-job.yaml:129]: checked and they fine as is and can't be fetched

**Review by furkatgofurov7 — DISMISSED** [review]
Thanks, this looks good to me.

**Review by Danil-Grigorev — DISMISSED** [review]

**Review by richardcase — APPROVED** [review]
The CI test is flaky but we can fix that later. I think this is good to go.

**Review by furkatgofurov7 — APPROVED** [review]

**Review by mbologna — APPROVED** [review]

**Pull request merged**